modules/biobambam/bamsormadup/main.nf

```diff
@@ -0,0 +1,44 @@
+process BIOBAMBAM_BAMSORMADUP {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::biobambam=2.0.183" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/biobambam:2.0.183--h9f5acd7_1':
+        'quay.io/biocontainers/biobambam:2.0.183--h9f5acd7_1' }"
+
+    input:
+    tuple val(meta), path(bam)
+    tuple val(meta), path(reference)
+
+    output:
+    tuple val(meta), path("*.{bam,cram}")       ,emit: bam
+    tuple val(meta), path("*.bam.bai")          ,optional:true, emit: bam_index
+    tuple val(meta), path("*.txt")              ,emit: metrics
+    path "versions.yml"                         ,emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def suffix = "cram" if args.contains("outputformat=cram") else "bam"
+
+    if (args.contains("outputformat=cram") and reference == null) error "Reference required for CRAM output."
```

Comment:
```suggestion
    if (args.contains("outputformat=cram") && reference == null) error "Reference required for CRAM output."
```

Normally the && operator is used AFAIK

Reply:
My bad. Not used to groovy yet I suppose 😅

Reply:
Doens't actually matter, I'm just so used to it it caught my eye!